src/Voron/Data/CompactTrees/CompactTree.Iterator.cs

```diff
@@ -66,13 +67,25 @@ public bool MoveNext(out Slice key, out long value)
 
                 return next;
             }
-
+            
             public bool MoveNext(out Span<byte> key, out long value)
             {
                 ref var state = ref _cursor._stk[_cursor._pos];
                 while (true)
                 {
-                    Debug.Assert(state.Header->PageFlags.HasFlag(CompactPageFlags.Leaf));
```

Comment:
when will this _not_ be the case here?

Reply:
When you are doing Seek and there is only one element with this prefix (so its leaf)